elide-core/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -263,12 +235,234 @@ private Object fetchRelationship(Environment context) {
         }
     }
 
-    /** stub code **/
-    private Object replaceObjects(Environment context) { return null; }
+    /**
+     * handle UPSERT operation
+     * @param context Environment encapsulating graphQL's request environment
+     * @return list of {@link PersistentResource} objects
+     */
+    private Set<PersistentResource> upsertObjects(Environment context) {
+        /* sanity check for id and data argument w UPSERT */
+        if(context.ids.isPresent()) {
+            throw new BadRequestException("UPSERT must not include ids");
+        }
+
+        if(!context.data.isPresent()) {
+            throw new BadRequestException("UPSERT must include data argument");
+        }
+
+        Class<?> entityClass;
+        EntityDictionary dictionary = context.requestScope.getDictionary();
+        if(context.isRoot()) {
+            entityClass = dictionary.getEntityClass(context.field.getName());
+        } else {
+            entityClass = dictionary.getParameterizedType(context.parentResource.getResourceClass(), context.field.getName());
+        }
+
+        /* form entities */
+        Set<Entity> entitySet = new HashSet<>();
+        for(Map<String, Object> input : context.data.get()) {
+            entitySet.add(new Entity(Optional.empty(), input, entityClass, context.requestScope));
```

Comment:
I still think you should put the parent here (if it exists) for correctness.  Otherwise, this might be a latent bug.